### PR TITLE
Internal: Use templates to setup tests in Core [ED-9488]

### PR DESCRIPTION
**Pull request by yotamselementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 
 test.describe( 'Nested Tabs tests @nested-tabs', () => {
+	let pageId;
+
+	test.beforeEach( async () => {
+		pageId = await createPage();
+	} );
+
+	test.afterEach( async () => {
+		pageId = await deletePage( pageId );
```
Comment (Nevoss) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:14]:
```suggestion
		await deletePage( pageId );
```

File: tests/playwright/sanity/bg-lazyload.test.js [hunk 2]
```diff
@@ -1,47 +1,67 @@
...
-		lazyloadElement.scrollIntoView();
-		lazyloadElement.setAttribute( 'bg-test-element', '' );
-	}, lazyloadSelector );
+	test.afterEach( async () => {
+		await deletePage( pageId );
+	} );
 
-	await page.waitForTimeout( 1500 );
+	test( 'Background lazy load sanity test', async ( { context, page } ) => {
+		const editorPage = new EditorPage( page, { config: { projects: [ { use: {} } ] } } );
```
Comment (Nevoss) [inline tests/playwright/sanity/bg-lazyload.test.js:36]:
```suggestion
		const editorPage = new EditorPage( page );
```

 I think we should remove the use of `testInfo` inside EditorPage, if you don't want to do so in this PR just use `testInfo`

```suggestion
		const editorPage = new EditorPage( page, testInfo );
```

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 3]
```diff
@@ -662,19 +627,15 @@ test.describe( 'Nested Tabs tests @nested-tabs', () => {
 		// Arrange.
 		const wpAdmin = new WpAdminPage( page, testInfo );
 		await setup( wpAdmin );
```
Comment (Nevoss) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:629]: What about this section? which activates and deactivates the experience for each test.

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 3 above)
Comment (yotamselementor) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:629]: Seems like a separate PR to me - this is an anti-pattern I see in a lot of tests

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 1 above)
Comment (yotamselementor) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Fixed in https://github.com/elementor/elementor/pull/21063/commits/3ccf1d714be4a48cf2fc2db219fb0b33b034c384

File: tests/playwright/sanity/bg-lazyload.test.js (hunk 2 above)
Comment (yotamselementor) [reply tests/playwright/sanity/bg-lazyload.test.js]: Fixed in https://github.com/elementor/elementor/pull/21063/commits/3ccf1d714be4a48cf2fc2db219fb0b33b034c384

**Review by Nevoss — APPROVED** [review]

**Pull request merged**